packages/server/src/api/controllers/datasource.ts

```diff
@@ -254,9 +268,9 @@ export async function destroy(ctx: UserCtx) {
   if (datasource.type === dbCore.BUDIBASE_DATASOURCE_TYPE) {
     await destroyInternalTablesBySourceId(datasourceId)
   } else {
-    const queries = await db.allDocs(getQueryParams(datasourceId))
+    const queries = await db.allDocs<RowValue>(getQueryParams(datasourceId))
```

Comment:
Given the use of `getQueryParams`, would this not be `db.allDocs<Query>`?

I was thinking recently about whether it would be possible to use conditional types to map `DocumentType` to the returned type here, but haven't gotten any further than dreaming about it.

Reply:
No - as there isn't an `include_docs` in the request, there won't be a document returned. Instead its just fetching metadata about the documents.

Reply:
Ahh I totally missed that. My bad.